Allow user to customize the labels used for the hints

IndianBoy42 opened the pull request:
Check vim.g.ts_hint_textobject_keys and use that first. Iterate through the rest afterwards

I *think* the logic for counting is correct but there could easily be a off-by-one error, I am new to lua and one-indexing

IndianBoy42 committed:
Allow user to customize the labels used for the hints

mfussenegger reviewed (commented):
Could you revert the unrelated formatting changes?

I think I'd also prefer a tshs specific config table that doesn't use pollute global namespace.

For example:


```lua
config = {
  hint_keys = nil
}
```


And as user you'd configure it with:

```lua
require('tshs').config.hint_keys = '...'
```

IndianBoy42 commented:
I added the new config, and I also implemented skipping the hint_keys while going throught he a-z afterwards, i'm not super happy with the implementation, we could precompute keys_iter and cache into an array so it takes less time. but I think it will be rare to have so many hint_labels anyway

mfussenegger commented:
> I added the new config, and I also implemented skipping the hint_keys while going throught he a-z afterwards, i'm not super happy with the implementation, we could precompute keys_iter and cache into an array so it takes less time. but I think it will be rare to have so many hint_labels anyway

Pre computing them sounds good to me.
I'd also be up for changing the default values to what you proposed in the readme change.

IndianBoy42 commented:
Yeah the homerow based default is better imo

I can work on precomputing tomorrow, i think it can be merged as is as well, the bad case is not so bad and won't be hit very often

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________
From: Mathias Fußenegger ***@***.***>
Sent: Friday, July 9, 2021 12:02:48 AM
To: mfussenegger/nvim-ts-hint-textobject ***@***.***>
Cc: Anshuman MEDHI ***@***.***>; Author ***@***.***>
Subject: Re: [mfussenegger/nvim-ts-hint-textobject] Allow user to customize the labels used for the hints (#2)


I added the new config, and I also implemented skipping the hint_keys while going throught he a-z afterwards, i'm not super happy with the implementation, we could precompute keys_iter and cache into an array so it takes less time. but I think it will be rare to have so many hint_labels anyway

Pre computing them sounds good to me.
I'd also be up for changing the default values to what you proposed in the readme change.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/mfussenegger/nvim-ts-hint-textobject/pull/2#issuecomment-876560165>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABNUNQICCT755LV7OANHSYDTWXD2RANCNFSM47ZSKY6Q>.

mfussenegger commented:
Merged - but I ended up dropping the default change - after trying the change for a bit I realized I prefer the existing default better because I can better tell which node is larger or smaller if there are several nodes within a single line.
